### PR TITLE
[Teleport] Add Teleport backend configuration

### PR DESCRIPTION
**Pull request by Nuru**
## what
Configure backend storage and related roles and secrets for [Teleport](https://gravitational.com/teleport/)

## why
Create resources needed by Teleport

### CODE REVIEW

File: aws/teleport/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+output "teleport_kubernetes_namespace" {
+  value = "${var.kubernetes_namespace}"
+}
+
+output "teleport_auth_iam_role" {
+  value = "${aws_iam_role.teleport.name}"
+}
+
+output "teleport_cluster_state_dynamo_table" {
```
Comment (osterman) [inline aws/teleport/outputs.tf]: Standardize on dynamodb rather than dynamo as a new convention.

File: aws/teleport/main.tf [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+resource "aws_ssm_parameter" "teleport_audit_events_uri" {
+  name        = "${format(var.chamber_parameter_name, local.chamber_service, "teleport_audit_events_uri")}"
+  value       = "dynamodb://${module.teleport_backend.dynamodb_audit_table_id}"
+  description = "Teleport audite events storage URI"
+  type        = "String"
+  overwrite   = "true"
+}
+
```
Comment (osterman) [inline aws/teleport/main.tf]: Standardize on dynamodb rather than dynamo as a new convention.

Comment (Nuru) [reply aws/teleport/main.tf]: done

File: aws/teleport/outputs.tf (hunk 1 above)
Comment (Nuru) [reply aws/teleport/outputs.tf]: done

**Pull request merged**